backend_new/src/services/user.service.ts

```diff
@@ -0,0 +1,234 @@
+          },
+          {
+            email: {
+              contains: params.search,
+              mode: Prisma.QueryMode.insensitive,
+            },
+          },
+          {
+            listings: {
+              some: {
+                name: {
+                  contains: params.search,
+                  mode: Prisma.QueryMode.insensitive,
+                },
+              },
+            },
+          },
+        ],
+      });
+    }
+
+    if (!params.filter?.length) {
+      return {
+        AND: filters,
+      };
+    }
+
+    params.filter.forEach((filter) => {
+      if (UserFilterKeys.isPartner in filter) {
+        filters.push({
+          userRoles: {
+            isPartner: filter[UserFilterKeys.isPartner],
+          },
+        });
+      } else if (UserFilterKeys.isPortalUser in filter) {
+        if (filter[UserFilterKeys.isPortalUser]) {
+          if (user?.userRoles?.isAdmin) {
+            filters.push({
+              OR: [
+                {
+                  userRoles: {
+                    isPartner: true,
+                  },
+                },
+                {
+                  userRoles: {
+                    isAdmin: true,
+                  },
+                },
+                {
+                  userRoles: {
+                    isJurisdictionalAdmin: true,
+                  },
+                },
+              ],
+            });
+          } else if (user?.userRoles?.isJurisdictionalAdmin) {
+            filters.push({
+              OR: [
+                {
+                  userRoles: {
+                    isPartner: true,
+                  },
+                },
+                {
+                  userRoles: {
+                    isJurisdictionalAdmin: true,
+                  },
+                },
+              ],
+            });
+            filters.push({
+              jurisdictions: {
+                some: {
+                  id: {
+                    in: user?.jurisdictions?.map((juris) => juris.id),
+                  },
+                },
+              },
+            });
+          } else {
+            filters.push({
+              userRoles: {
+                isPartner: true,
+              },
+            });
+          }
+        } else {
+          filters.push({
+            AND: [
+              {
+                OR: [
+                  {
+                    userRoles: {
+                      isPartner: null,
+                    },
+                  },
+                  {
+                    userRoles: {
+                      isPartner: false,
+                    },
+                  },
+                ],
+              },
+              {
+                OR: [
+                  {
+                    userRoles: {
+                      isAdmin: null,
+                    },
+                  },
+                  {
+                    userRoles: {
+                      isAdmin: false,
+                    },
+                  },
```

Comment:
Are we missing adding filters for `isJurisdictionalAdmin` here?

Reply:
wow really good catch, thats a bug from the current typeorm backend
so one less bug being brought over!